app/models/deploy_metrics.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Used to collect metrics of cycle time(elapsed time needed to deliver a project to production)
+# Check out https://about.gitlab.com/2019/03/04/reduce-cycle-time/ for more details on cycle time
+# The main objective for this is to gain more insight on the efficiency of the end-to-end project development process
+# There are 2 key metrics:
+# 1. PR - Production: this will show the time needed for a pull request to get deployed to production environment
+# 2. Staging - Production: this will show the time needed to deliver from staging(non-production environment)
+#                          to production environment.
+class DeployMetrics
+  NOT_APPLICABLE = 0
+
+  def initialize(deploy)
+    @deploy = deploy
+  end
+
+  def cycle_time
+    if @deploy&.production && @deploy&.status == "succeeded"
+      {pr_production: pr_production, staging_production: staging_production}
+    else
+      {pr_production: NOT_APPLICABLE, staging_production: NOT_APPLICABLE}
```

Comment:
get rid of NOT_APPLICABLE and return {} here ... then for the reporting just report each key/value ... to get less repetition/code that needs to be in sync all the time
```suggestion
      {}
```